### DISCUSSION TITLE
Can OS windows be named?

### DISCUSSION THREAD

**Discussion by mfilej** [kovidgoyal/kitty#6067] [Q&A]
I see in the `kitty @ ls` output that there are `wm_name` and `wm_class` attributes, but for me, they always say `"kitty"`. Is there a way to change the name?

**Comment by kovidgoyal** [comment]
On Sun, Feb 26, 2023 at 11:50:40AM -0800, Miha Filej wrote:
> I see in the `kitty @ ls` output that there are `wm_name` and `wm_class` attributes, but for me, they always say `"kitty"`. Is there a way to change the name?

Run kitty with --name and --class or use th ecorresponding options in
the launch command.

**Comment by mfilej** [comment]
Would this allow me to rename an existing OS window?

I tried running the suggested command from a running kitty instance:
```
$ kitty --name foo
Unknown option: --name
$ kitty --class boo
Unknown option: --class
```

**Comment by page-down** [comment]
> ... they always say "kitty".

Under macOS these two are not used, so they are both `kitty`.

> I tried running the suggested command ...

Both options are not available under macOS.
But you can use `launch --os-window-name`.

> Is there a way to change the name?
> Would this allow me to rename an existing OS window?

Yes, write a kitten.
Here is an example of getting an OS window by id and renaming it (omitting the boilerplate code, and should be set as `no_ui=True`).
You can also find the corresponding OS window by name and change it.

```python
tm = boss.os_window_map.get(int(args[1]))
if tm is not None:
    tm.wm_name = args[2]
```

```shell
kitty @ kitten macos_rename_os_window.py 1 new-name
```

Then you can find the corresponding OS window id and the tabs in it according to the name, and finally create new tabs or other actions through the tabs in it.

It would be handy to be able to match the OS window directly and then create tabs in it.
